### PR TITLE
[Silabs] Fixing the build for the custom boards on NCP

### PR DESCRIPTION
**Pull request by chirag-silabs**
The spi files on NCP on the EFR32 was only building for defined boards i.e., BRD4187C and BRD4186C and not for the custom boards. 

Fixing the build if any custom board is provided during the build.

### CODE REVIEW

File: examples/platform/silabs/efr32/rs911x/hal/rsi_board_configuration.h [hunk 1]
```diff
@@ -35,7 +35,8 @@ typedef struct
 #elif defined(EFR32MG24_BRD4187C) || defined(BRD4187C)
 #include "brd4187c.h"
 #else
-#error "Need SPI Pins"
+#include "sl_custom_board.h"
+#warning "SPIDRV USART peripheral not configured correctly"
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/hal/rsi_board_configuration.h:39]: Is this warning message valid? Why do we allow this path? I could not find the content of sl_custom_board.h. How do we know that peripheral isn't correctly configured in in the sl_custom_board

File: examples/platform/silabs/efr32/wf200/sl_wfx_board.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #elif defined(EFR32MG24_BRD4187C) || defined(BRD4187C) || defined(EFR32MG24_BRD4187A) || defined(BRD4187A)
 #include "brd4187c.h"
 #else
-#error "Need SPI Pins"
+#include "sl_custom_board.h"
+#warning "SPIDRV USART peripheral not configured correctly"
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/wf200/sl_wfx_board.h:29]: same comment here

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/platform/silabs/efr32/rs911x/hal/rsi_board_configuration.h (hunk 1 above)
Comment (chirag-silabs) [reply examples/platform/silabs/efr32/rs911x/hal/rsi_board_configuration.h]: Updated the warning message

File: examples/platform/silabs/efr32/sl_custom_board.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+#define PIN_OUT_CLEAR 0
+
+#define MY_USART USART0
+#define MY_USART_CLOCK cmuClock_USART0
+#define MY_USART_TX_SIGNAL dmadrvPeripheralSignal_USART0_TXBL
+#define MY_USART_RX_SIGNAL dmadrvPeripheralSignal_USART0_RXDATAV
+
+#define SL_WFX_HOST_PINOUT_RESET_PORT gpioPortA
+#define SL_WFX_HOST_PINOUT_RESET_PIN 5
+#define SL_WFX_HOST_PINOUT_SPI_WIRQ_PORT gpioPortA /* SPI IRQ port*/
```
Comment (yufengwangca) [inline examples/platform/silabs/efr32/sl_custom_board.h:49]:
```suggestion
#define SL_WFX_HOST_PINOUT_SPI_WIRQ_PORT gpioPortA /* SPI IRQ port */
```

File: examples/platform/silabs/efr32/sl_custom_board.h [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+#define MY_USART EUSART1
+#define MY_USART_CLOCK cmuClock_EUSART1
+#define MY_USART_TX_SIGNAL dmadrvPeripheralSignal_EUSART1_TXBL
+#define MY_USART_RX_SIGNAL dmadrvPeripheralSignal_EUSART1_RXDATAV
+
+#define WFX_RESET_PIN PIN(A, 6)
+#define WFX_INTERRUPT_PIN PIN(A, 7)
+#ifdef EXP_BOARD
+#define WFX_SLEEP_CONFIRM_PIN PIN(D, 2) /* Exp hdr 7 */
```
Comment (yufengwangca) [inline examples/platform/silabs/efr32/sl_custom_board.h:31]: Are these different pins both for Exp hdr 7?

File: examples/platform/silabs/efr32/sl_custom_board.h [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+#define EUS1CS_PIN 0
+
+#define MY_USART EUSART1
+#define MY_USART_CLOCK cmuClock_EUSART1
+#define MY_USART_TX_SIGNAL dmadrvPeripheralSignal_EUSART1_TXBL
+#define MY_USART_RX_SIGNAL dmadrvPeripheralSignal_EUSART1_RXDATAV
+
+#define WFX_RESET_PIN PIN(A, 6)
+#define WFX_INTERRUPT_PIN PIN(A, 7)
+#ifdef EXP_BOARD
```
Comment (yufengwangca) [inline examples/platform/silabs/efr32/sl_custom_board.h:30]: Are those different pin defines both for Exp hdr 7?

**Review by yufengwangca — APPROVED** [review]

File: examples/platform/silabs/efr32/sl_custom_board.h (hunk 5 above)
Comment (jmartinez-silabs) [reply examples/platform/silabs/efr32/sl_custom_board.h:30]: no, will remove the comment for the non expansion board

File: examples/platform/silabs/efr32/sl_custom_board.h [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+#define WFX_SLEEP_CONFIRM_PIN PIN(D, 2) /* Exp hdr 7 */
+#else
+#define WFX_SLEEP_CONFIRM_PIN PIN(A, 5) /* Exp hdr 7 */
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/sl_custom_board.h:33]:
```suggestion
#define WFX_SLEEP_CONFIRM_PIN PIN(D, 2) /* Expansion header pin7 */
#else
#define WFX_SLEEP_CONFIRM_PIN PIN(A, 5)
```

**Pull request merged**